### PR TITLE
Extended ecosystem check with scraped data

### PR DESCRIPTION
**Pull request by konstin**
This allows running the ecosystem CI locally in an isolated docker container using the data from https://github.com/akx/ruff-usage-aggregate. It also changes the output format of the ecosystem CI a little to include the repository URL.

CC @akx @sciyoshi i merged your efforts in one locally runnable docker container

### CODE REVIEW

**Review by akx — COMMENTED** [review]
Cool! 😎

**Comment by akx** [comment]
I just refreshed https://github.com/akx/ruff-usage-aggregate/blob/master/data/known-github-tomls.jsonl with about 90 new repos.

File: scripts/Dockerfile.ecosystem [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# From the project root:
+# ```
+# cargo build
+# docker buildx build -f scripts/Dockerfile.ecosystem -t ruff-ecosystem-checker --load .
+# docker run --rm ruff-ecosystem-checker
+# ```
+
+FROM python:3.11
```
Comment (MichaReiser) [inline scripts/Dockerfile.ecosystem:8]: What's your reasoning behind running it in a docker container vs running the script locally?

File: scripts/check_ecosystem.py [hunk 2]
```diff
@@ -46,14 +52,22 @@ async def clone(self: Self) -> "AsyncIterator[Path]":
...
                 "1",
                 "--no-tags",
-                "--branch",
-                self.ref,
-                f"https://github.com/{self.org}/{self.repo}",
-                tmpdir,
+            ]
+            if self.ref:
+                git_command.extend(["--branch", self.ref])
+            git_command.extend(
```
Comment (MichaReiser) [inline scripts/check_ecosystem.py:58]: Nit: Add a newline to make it clearer that this statement does not belong to the if body?
```suggestion
                git_command.extend(["--branch", self.ref])

          	git_command.extend(
```

**Review by MichaReiser — APPROVED** [review]
Nice!

Could we add a section to our Contribution.md that explains how to run the ecosystem checks?

File: scripts/Dockerfile.ecosystem (hunk 1 above)
Comment (konstin) [reply scripts/Dockerfile.ecosystem:8]: https://moyix.blogspot.com/2022/09/someones-been-messing-with-my-subnormals.html or rather, running software not designed around working with arbitrary untrusted inputs with the largest set of untrusted inputs i could find

**Comment by konstin** [comment]
> Could we add a section to our Contribution.md that explains how to run the ecosystem checks?

done

**Pull request merged**